charmcraft/logsetup.py

```diff
@@ -76,10 +80,19 @@ def set_mode(self, mode):
 
     def _set_filehandler(self):
         """Set the file handler to log everything to the temp file."""
-        _, self._log_filepath = tempfile.mkstemp(prefix="charmcraft-log-")
+        managed_mode = is_charmcraft_running_in_managed_mode()
+        if managed_mode:
+            self._log_filepath = str(get_managed_environment_log_path())
+        else:
+            _, self._log_filepath = tempfile.mkstemp(prefix="charmcraft-log-")
+
+        file_handler = logging.FileHandler(self._log_filepath, mode="w")
+
+        log_format = FORMATTER_DETAILED
+        if managed_mode:
+            log_format = "mm " + log_format
```

Comment:
I would probably make this refer to a constant

```
FORMATTER_DETAILED_MANAGED = "%(asctime)s mm %(name)-40s %(levelname)-8s %(message)s"
```

Reply:
Agree with Sergio that using this constant would be better.

Reply:
Done!